### PR TITLE
Fix or suppress SAT CompareObjectsWithEquals findings

### PR DESCRIPTION
**Pull request by wborn**
They are errors in SAT 0.12.0.

### CODE REVIEW

**Review by ssalonen — COMMENTED** [review]
Review of fmiweather

File: bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/phonebook/PhonebookProfile.java [hunk 1]
```diff
@@ -139,7 +139,7 @@ public void onStateUpdateFromHandler(State state) {
         if (state instanceof StringType) {
             Optional<String> match = resolveNumber(state.toString());
             State newState = match.map(name -> (State) new StringType(name)).orElse(state);
-            if (newState == state) {
+            if (newState.equals(state)) {
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/phonebook/PhonebookProfile.java:142]: Can you please revert this too and add the `"PMD.CompareObjectsWithEquals"` suppression? Thanks.

Comment (wborn) [reply bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/phonebook/PhonebookProfile.java]: I've addressed the comment.

File: bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/AbstractWeatherHandler.java [hunk 2]
```diff
@@ -94,7 +95,7 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     logger.trace("REFRESH received. Delaying by {} ms to avoid throttle excessive REFRESH",
                             delayRemainingMillis);
                 }
-                if (prevFuture == newFuture) {
```
Comment (ssalonen) [inline bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/AbstractWeatherHandler.java:97]: Adding this comment again as github seemingly lost it after Force push.

This has been intentionally comparing by identity.

**Review by ssalonen — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/AbstractWeatherHandler.java (hunk 2 above)
Comment (wborn) [reply bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/AbstractWeatherHandler.java:97]: Addressed as well, now I see the comment. :wink:

Comment (ssalonen) [reply bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/AbstractWeatherHandler.java:97]: Thanks! Good to have the comment  👌

**Review by ssalonen — APPROVED** [review]
Lgtm for fmiweather

**Review by cweitkamp — APPROVED** [review]
No further objections. I will merge it now.

LGTM. Thanks.

**Pull request merged**